nixos/modules/services/security/fail2ban.nix

```diff
@@ -3,23 +3,46 @@
 with lib;
 
 let
-
   cfg = config.services.fail2ban;
 
-  fail2banConf = pkgs.writeText "fail2ban.local" cfg.daemonConfig;
+  settingsFormat = pkgs.formats.keyValue { };
 
-  jailConf = pkgs.writeText "jail.local" ''
-    [INCLUDES]
+  configFormat = pkgs.formats.ini {
+    mkKeyValue = generators.mkKeyValueDefault { } " = ";
+  };
 
-    before = paths-nixos.conf
+  mkJailConfig = name: attrs:
+    optionalAttrs (name != "DEFAULT") { inherit (attrs) enabled; } //
```

Comment:
Maybe we should have `jails.DEFAULT.enabled` as an assertion, just in case someone tries to disable it.

Reply:
Or better yet: `readOnly = name == "DEFAULT";` in the `enabled` option (where `name` is taken from the submodule arguments).